xtraplatform-string-templates/src/test/groovy/de/ii/xtraplatform/stringtemplates/domain/StringTemplateFiltersSpec.groovy

```diff
@@ -0,0 +1,170 @@
+package de.ii.xtraplatform.stringtemplates.domain
+
+import spock.lang.Ignore
+import spock.lang.Specification
+
+class StringTemplateFiltersSpec extends Specification {
+
+    @Ignore
+    def 'All arguments are null'() {
+        when:
+            String result = StringTemplateFilters.applyTemplate(null, null, null, null)
+        then:
+            result.isEmpty()
+    }
+
+    @Ignore
+    def 'Template is null, value is present'() {
+        given:
+            String template = null
+            String value = "foobar"
+        when:
+            String result = StringTemplateFilters.applyTemplate(template, value)
+        then:
+            result.isEmpty()
+    }
+
+    @Ignore
+    def 'Value is null, template is present'() {
+        given:
+            String template = "{{value | replace:'ABC':' '}}"
+            String value = null
+        when:
+            String result = StringTemplateFilters.applyTemplate(template, value)
+        then:
+            result.isEmpty()
+    }
+
+    @Ignore
+    def 'Template is empty, value is present'() {
+        given:
+            String template = ""
+            String value = "foobar"
+        when:
+            String result = StringTemplateFilters.applyTemplate(template, value)
+        then:
+            result == value
+    }
+
+    def 'Value is empty, template is present'() {
+        given:
+            String template = "{{value | replace:'ABC':' '}}"
+            String value = ""
+        when:
+            String result = StringTemplateFilters.applyTemplate(template, value)
+        then:
+            result.isEmpty()
+    }
+
+    @Ignore
+    def 'Malformed template'() {
```

Comment:
I think in this case an empty result (that I assume the current implementation returns) should be fine. The empty result might serve as a hint for the user that the template is malformed.

Reply:
The current implementation returns the value of the template. Should I update it to return an empty string?